### PR TITLE
Add CMake Export files to install step in IlmBase and OpenEXR

### PR DESCRIPTION
**Pull request by hjmallon**
I am beginning work on adding ilmbase and openexr to hunter. See https://github.com/ruslo/hunter/issues/361
If this were to be merged then I might be able to avoid forking the repo as it is needed by some CMake packaging tools.

This CMake package support is detailed here https://cmake.org/cmake/help/v3.2/manual/cmake-packages.7.html#creating-packages and inspiration was mainly taken from here https://github.com/google/benchmark

### CODE REVIEW

**Comment by hjmallon** [comment]
Reset to WIP as still testing.

**Comment by hjmallon** [comment]
I am now happy with this. With this plus a couple of other changes (which I can send separately) I was able to get IlmBase and OpenEXR to build with hunter.

EDIT: I have updated again below. I really could do with someone who knows this CMake packaging stuff well to give this a review.

**Comment by hjmallon** [comment]
This is similar to #167 and #178 (but I have made fewer sweeping changes for now, they could come in a second PR if required). Maybe @AWhetter might be interested to see it?

**Comment by AWhetter** [comment]
Very nice! I would love to see this merged if my pull requests get rejected for whatever reason.

**Comment by hjmallon** [comment]
I agree with @AWhetter that his more fully featured Pull Req is preferable.

File: IlmBase/Config.cmake.in [hunk 1]
```diff
@@ -0,0 +1 @@
+include("${CMAKE_CURRENT_LIST_DIR}/@TARGETS_EXPORT_NAME@.cmake")
```
Comment (ruslo) [inline IlmBase/Config.cmake.in:1]: From [documentation](https://cmake.org/cmake/help/latest/module/CMakePackageConfigHelpers.html#generating-a-package-configuration-file):
```
check_required_components(<PackageName>) should be called at the end of the
FooConfig.cmake file ... This macro should be called even if the package
doesn’t provide any components to make sure users are not specifying components
erroneously.
```

Example:
- https://github.com/forexample/package-example/blob/00335ff3f39c3b82a4b4b6095412b5a9e3aeda81/Foo/cmake/Config.cmake.in

File: IlmBase/Half/CMakeLists.txt [hunk 2]
```diff
@@ -67,11 +67,12 @@ IF (BUILD_ILMBASE_STATIC)
...
 
 IF (OPENEXR_BUILD_SHARED OR BUILD_ILMBASE_STATIC)
-  INSTALL ( TARGETS
-    ${ILMBASE_LIB_TARGETS}
+  INSTALL ( TARGETS ${ILMBASE_LIB_TARGETS}
+      EXPORT "${TARGETS_EXPORT_NAME}"
       ARCHIVE DESTINATION lib
       LIBRARY DESTINATION lib
       RUNTIME DESTINATION ${RUNTIME_DIR}
+      INCLUDES DESTINATION include/OpenEXR
```
Comment (ruslo) [inline IlmBase/Half/CMakeLists.txt:75]: This one will modify `INTERFACE_INCLUDE_DIRECTORIES`. I'm not sure about C++ part but usually it's just `include`, e.g.:
- https://cmake.org/cmake/help/latest/manual/cmake-packages.7.html#creating-packages

**Review by ruslo — CHANGES_REQUESTED** [review]

**Comment by hjmallon** [comment]
Thanks for the review @ruslo . I think something based on @AWhetter 's version with a more complete overhaul is probably more desirable.

**Comment by kdt3rd** [comment]
I have completely refactored the cmake infrastructure for openexr to use the "modern cmake" paradigm. If you could take a look at the current master, and see if I missed something about the config mechanism, would be much appreciated.

**Comment by kdt3rd** [comment]
The complete refactor of the cmake build has been completed and merged onto master. If there are any issues with the new export mechanism, please report them as bugs / new PRs. Thanks for all the contributions!

**Pull request closed without merging**